src/main/java/core/basesyntax/storage/FruitTransaction.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.storage;
+
+public class FruitTransaction {
+    private String fruit;
+    private int quantity;
+    private Operation operation;
+
+    public FruitTransaction() {
+    }
+
+    public FruitTransaction(String fruit, int quantity) {
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getByCode(String character) {
+            for (Operation operation : values()) {
+                if (operation.getCode().equals(character)) {
+                    return operation;
+                }
+            }
+            throw new RuntimeException("Unknown operation :" + character);
```

Comment:
```suggestion
            throw new IllegalArgumentException("Unknown operation: " + code);
```
it's better to throw more specific exceptions where possible